### PR TITLE
Bugfix for monkey without -s flag

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature / Fixes
> Fixes bug where monkey crashes if default server flag is not passed

* [X] Have you successfully tested your changes locally?

### CODE REVIEW

File: monkey/infection_monkey/monkey.py [hunk 1]
```diff
@@ -116,9 +114,6 @@ def start(self):
             monkey_tunnel.start()
 
         StateTelem(False).send()
-
-        self._default_server = WormConfiguration.current_server
```
Comment (VakarisZ) [inline monkey/infection_monkey/monkey.py]: My only concern is that I moved this code to execute earlier and overlooked code that will change `WormConfiguration.current_server`, but it doesn't seem so.

**Comment by danielguardicore** [comment]
Reminds me to ask, why are we using the default port here?

`machine.set_default_server(get_interface_to_target(machine.ip_addr) +                                                  (':'+self._default_server_port if self._default_server_port else ''))`

Why are we doing that and not passing in the "real" server IP and if that fails, using the tunnel?

**Comment by VakarisZ** [comment]
What do you mean by "real" server IP? We pass the one that monkey can reach. If island is listening on 192.168.56.1 and 10.5.5.5 and we exploit a host that's 10.5.5.6, we need to pass 10.5.5.5:5000 as a default server, not 192.168.56.1:5000. How is tunneling gona help here?

**Comment by danielguardicore** [comment]
`get_interface_to_target` returns the attacker machines interface/IP towards the victim, not the server IP.

**Comment by VakarisZ** [comment]
Server is an attacker machine if monkey is launched on it.

**Comment by danielguardicore** [comment]
And if it's not...

**Comment by VakarisZ** [comment]
if self._network.on_island(self._default_server):
                        machine.set_default_server(get_interface_to_target(machine.ip_addr) +
                                                   (':'+self._default_server_port if self._default_server_port else ''))
                    else:
                        machine.set_default_server(self._default_server)

**Comment by danielguardicore** [comment]
🤦‍♂️

**Pull request merged**